libs/ui/search/src/lib/record-table/record-table.component.ts

```diff
@@ -32,4 +40,38 @@ export class RecordTableComponent {
     }
     return undefined
   }
+
+  getRecordFormats(record: CatalogRecord): FileFormat[] {
+    if (record.kind === 'service') {
+      return []
+    }
+    const formatsAndPrio = record.distributions.map(
+      (distribution: DatasetDistribution) => {
+        return {
+          format: getFileFormat(distribution),
+          priority: sortPriority(distribution),
+        }
+      }
+    )
+
+    const sortedFormats = formatsAndPrio
+      .sort((a, b) => a.priority - b.priority)
+      .map((element) => element.format)
+
+    const types = new Set(sortedFormats)
+    return Array.from(types).filter((elm) => elm)
+  }
+
+  getBadgeColor(format: FileFormat): string {
+    return getBadgeColor(format)
+  }
+
+  firstFormat(record: CatalogRecord): FileFormat {
+    return this.getRecordFormats(record)[0]
+  }
+
+  secondToLastFormat(record: CatalogRecord): FileFormat[] {
+    const formats = this.getRecordFormats(record)
+    return formats.slice(formats.length - 2)
+  }
```

Comment:
There is I think an issue with these functions: using either of them will recompute the formats array each time, and this will happen many times because they are used several times in the template. I usually don't worry too much about recomputing values on the fly, but in that case it feels a little too much.

One solution would be to only keep the `getRecordFormats` function, and store its result in the template using the `ngIf` directive, e.g.:

```html
<div class="record-table-col text-16 basis-1/2">
  <div *ngIf="getRecordFormats(record) as formats">
    <span
        class="flex badge-btn text-sm text-white"
        [style.background-color]="getBadgeColor(formats[0])"
    >{{ records[0] }}</span>
  </div>
</div>
```

Then the equivalent of `secondToLastFormat` would simply be `formats.slice(1)`.